### DISCUSSION TITLE
Transparent gradient not giving expected colors

### DISCUSSION THREAD

**Discussion by MMz10000** [ImageMagick/ImageMagick#2080] [Help]
Good evening,

My portable IM version is ImageMagick 7.0.10-15 Q16 x64 2020-05-30 running on windows 10 x64 (using powershell)

Using the following script `magick -size 256x256 gradient:"#FF0000FF"-"#0000FF00" gradient.png` gives unexpected results.

Here is the current output:
![gradient](https://user-images.githubusercontent.com/47039265/83338633-3121b500-a2be-11ea-964a-8727e1b0f840.png)
And here is the same picture under GIMP, with the alpha channel disabled:
![image](https://user-images.githubusercontent.com/47039265/83338658-71813300-a2be-11ea-98f6-a14ebf72ff16.png)

Here is what was expected (generated under GIMP):
![gradientb](https://user-images.githubusercontent.com/47039265/83338671-8bbb1100-a2be-11ea-974c-923c816cd04d.png)
And the screenshot under GIMP, with the alpha disabled:
![image](https://user-images.githubusercontent.com/47039265/83338678-9bd2f080-a2be-11ea-86ed-815ff4370112.png)

The blue component as been ignored while generating the gradient.

Could you help me to understand and fix this please?

**Comment by fmw42** [comment]
It looks like ImageMagick is doing alpha blending and GIMP is not. I am not sure if this is a bug or just the way ImageMagick handles gradients with alpha. If you want the same behavior with gradients, you can process the base image separately from the alpha channel and then put the gradient alpha channel back into the image.

**One of the IM developers will have to comment further on whether this is the way IM handles gradients with transparency or if there is a more direct and simple way to force no doing alpha blending when creating gradients with transparency.**

*Perhaps we need a new -define gradient:alpha-blending=off*

**Reply by MMz10000** [reply]
Thanks again @fmw42 ,

Is there an option to tell IM to not consider the alpha of the color I'm passing through, like: `magick -size 256x256 -mysteriousOption gradient:"#FF0000FF"-"#0000FF00" gradient_noAlpha.png`

Or do I need to process the strings beforehand and send the good colors: `magick -size 256x256 gradient:"#FF0000"-"#0000FF" gradient_noAlpha.png`

**Reply by fmw42** [reply]
It is just -alpha off and it has to go after creating the gradient.

`magick -size 256x256 gradient:"#FF0000FF"-"#0000FF00" -alpha off gradient_noAlpha.png`

**Reply by MMz10000** [reply]
Thanks a lot again @fmw42 ,

I guess if it was that simple, I should be able to find the solution to do the opposite (just the alpha) myself. I'm really grateful for your patience and help you provided me today.

IM is a very interesting software. It is a lot more enjoyable to learn how to use it with such a helpful community behind.

**Reply by MMz10000** [reply]
So I unmarked as answer because after testing, I can see the blue component as been ignored again... I guess the problem is within the `gradient:`.

Here is the output of `magick -size 256x256 gradient:"#FF0000FF"-"#0000FF00" -alpha off gradient_noAlpha.png`:
![gradient](https://user-images.githubusercontent.com/47039265/83339713-6df2a980-a2c8-11ea-8a59-4b3fb9f274d3.png)

No worries, as my list of colors is part of a function, I could easily add a new parameter with a list of Alphas, and use a list of Colors without alpha components in them.

**Comment by fmw42** [comment]
Try this. 

```
magick -size 256x256 gradient:"#FF0000"-"#0000FF" gradient_noAlpha.png
magick -size 256x256 gradient:"#FFFFFF"-"#000000" gradient_Alpha.png
magick gradient_noAlpha.png gradient_Alpha.png -alpha off -compose copy_opacity -composite result.png

```

**Reply by MMz10000** [reply]
Here is the output:
![result](https://user-images.githubusercontent.com/47039265/83340087-02124000-a2cc-11ea-8608-29429de924e9.png)

**Reply by fmw42** [reply]
OK. But if you now remove the alpha channel, does it look the way you want as in GIMP?

**Reply by MMz10000** [reply]
Yes, this is the expected result.
This is what I will achieve changing the function as GenerateRadialGradient ( OutputName, Colors[], Offsets[], (optional)Alphas[])

This is what I tried to explain there: https://github.com/ImageMagick/ImageMagick/discussions/2080#discussioncomment-20488

I will also have to handle the different colors IM accepts, such as "yellow", "rgb(r,g,b)", and so on... But nothing a good switch cannot process :D
